app/routes.py

```diff
@@ -1,2 +1,231 @@
-from flask import Blueprint
+from app import db
+from app.models.task import Task
+from app.models.goal import Goal
+from flask import Blueprint, jsonify, request, abort, make_response
+from datetime import datetime, timezone
+import os
+from dotenv import load_dotenv
+import requests
 
+tasks_bp = Blueprint("task_list", __name__, url_prefix="/tasks")
+
+def valid_int(number, parameter_type):
+    try:
+        int(number)
+    except:
+        abort(make_response({"error": f"{parameter_type} must be an int"})), 400
+
+def get_task_from_id(task_id):
+    valid_int(task_id, "task_id")
+    return Task.query.get_or_404(task_id, description="{task not found}")
+
+@tasks_bp.route("", methods=["POST"])
+def post_new_task():
+    
+    request_body = request.get_json()
+    if "title" not in request_body or "description" not in request_body\
+         or "completed_at" not in request_body:
+        return jsonify({"details": "Invalid data"}), 400
+
+    new_task = Task(title=request_body["title"],
```

Comment:
Consider encapsulating this functionality in a helper method that either aborts for invalid data or returns the `new_task` if the request body is valid